Report for GA release

LeoMcA opened the pull request:
https://github.com/mozilla/sumo-project/issues/552

My first pass at this, in case you wanted to review my general approach before I continue tomorrow @akatsoulas.

Outstanding things:
* ~Add tests to verify `submitted_to_play_store_at` is set automatically~
* Post metrics to slack instead of just printing them
* ~Verify these are in fact the metrics we want to gather~

akatsoulas commented on mresponse/utils/management/commands/gather_metrics.py:
```diff
@@ -0,0 +1,161 @@
+        return reviews.count()
+
+    def responded_reviews(
+        self, language="en", period=timezone.timedelta(hours=72), since=None
```
Var for language and hours

akatsoulas commented on mresponse/utils/management/commands/gather_metrics.py:
```diff
@@ -0,0 +1,161 @@
+        review_count = 0
+        responded_in_period_count = 0
+
+        for review in reviews.iterator():
```
Interesting choice to alter the default behavior. May I ask why this is needed?

LeoMcA replied:
As discussed out of band, I'm concerned about us OOMing if we have a lot of reviews to consider. Good discussion though, made me realising we were N+1ing here. A `select_related` fixes that for us.

akatsoulas commented on mresponse/utils/management/commands/gather_metrics.py:
```diff
@@ -0,0 +1,161 @@
+            ):
+                responded_in_period_count += 1
+
+        if review_count == 0:
```
nit: you can reduce this to 

```python
return responed_in_period_count / review_count if review_count else 0
```

akatsoulas commented on mresponse/utils/management/commands/gather_metrics.py:
```diff
@@ -0,0 +1,161 @@
+
+    def active_contributors(
+        self,
+        required_responses=5,
```
vars here too

akatsoulas commented on mresponse/utils/management/commands/gather_metrics.py:
```diff
@@ -0,0 +1,161 @@
+        self,
+        required_responses=5,
+        required_moderations=15,
+        period=timezone.timedelta(days=7),
```
Calculate the timedelta in the body of the function instead of here. If you change this to `days=DAYS` where DAYS=7 and then you change the var, you will get surprising results.

akatsoulas commented on mresponse/utils/management/commands/gather_metrics.py:
```diff
@@ -0,0 +1,161 @@
+
+        for response in responses.iterator():
+            author = response.author_id
+            if author in users:
```
Won't this check fail b/c of `users={}`?

akatsoulas commented on mresponse/utils/management/commands/gather_metrics.py:
```diff
@@ -0,0 +1,161 @@
+            else:
+                users[author] = {"responses": 1, "moderations": 0}
+
+        for moderation in moderations.iterator():
```
Instead of looping through the objects you can use Aggregations to get the numbers you are looking for directly from the query.

LeoMcA commented on mresponse/utils/management/commands/gather_metrics.py:
```diff
@@ -0,0 +1,161 @@
+
+
+class Command(BaseCommand):
+    def fetched_reviews(self, period=timezone.timedelta(hours=24)):
```
Since we set the defaults through the add_arguments function anyway, I made all of these function defaults a type-appropriate "null" value

LeoMcA commented:
Rebased against #506 and updated in another branch: https://github.com/LeoMcA/m-response/commit/fb4091e94768cd0de3a2b47704e2a0ba0ec820aa

Will update this PR once #506 is merged.

akatsoulas commented on mresponse/utils/management/commands/gather_metrics.py:
```diff
@@ -0,0 +1,148 @@
+
+        for review in reviews:
+            review_count += 1
+            response = review.responses.first()
```
Why are you using `first()` here?

akatsoulas replied:
Or review.responses.filter(rejected=False)

LeoMcA replied:
As discussed out of band, so I don't need to catch any `.get()` exceptions. Also I don't expect multiple submitted reviews to happen, although it's a theoretical possibility.

akatsoulas commented on mresponse/utils/management/commands/gather_metrics.py:
```diff
@@ -0,0 +1,148 @@
+                responses_count__gte=required_responses,
+                moderations_count__gte=required_moderations,
+            )
+            .count()
```
not related to the code but this looks like a weird metric. I would expect that the number of moderations and the number of responses per user would make more sense.
